test/unittests/testutil.cc

```diff
@@ -245,7 +247,55 @@ MockCatalog* MockCatalog::AttachFreely(const std::string  &root_path,
   return new_catalog;
 }
 
-void MockCatalog::RegisterChild(MockCatalog *child) {
+void MockCatalog::RemoveChild(MockCatalog *child) {
```

Comment:
Ok, thanks!  Just add a comment to each one of the functions that explains which of the list the function modifies.